src/sentry/api/endpoints/project_platforms.py

```diff
@@ -11,8 +11,22 @@ class ProjectPlatformSerializer(Serializer):
     def serialize(self, obj, attrs, user):
         return {'platform': obj.platform, 'dateCreated': obj.date_added}
 
+ERR_FIELD_REQUIRED = 'This field is required.'
+
 
 class ProjectPlatformsEndpoint(ProjectEndpoint):
     def get(self, request, project):
         queryset = ProjectPlatform.objects.filter(project_id=project.id)
         return Response(serialize(list(queryset), request.user))
+
+    def put(self, request, project):
+        platform_name = request.DATA.get('platform')
+
+        if platform_name is None:
```

Comment:
This should go through an input Serializer instead to automatically do this validation. This serializer then can also handle an enum of values. I'm not sure this is good idea to be entirely open ended, so we probably need to lock it down to a  finite set of choices.

Reply:
There is a finite set of choices: `src/sentry/integration_docs/_platforms.json`

The hard thing is that this is generated by the build process, so tests are a little weirder. I forget exactly, but @MaxBittker and @denamwangi ran into this exact thing when building out the signup flow this summer. They solved it for the frontend case, not sure if we solved it for backend.